src/Raven.Server/Documents/Handlers/QueriesHandler.cs

```diff
@@ -192,6 +201,30 @@ private IndexQueryServerSide GetIndexQuery(DocumentsOperationContext context, Ht
             return indexQuery;
         }
 
+        private void Suggestions(DocumentsOperationContext context, string indexName, OperationCancelToken token)
+        {
+            var existingResultEtag = GetLongFromHeaders("If-None-Match");
+
+            var query = SuggestionsQueryServerSide.Create(HttpContext, GetPageSize(), context);
+            var runner = new QueryRunner(Database, context);
+
+            var result = runner.ExecuteSuggestionsQuery(indexName, query, context, existingResultEtag, token);
+            if (result.NotModified)
+            {
+                HttpContext.Response.StatusCode = (int)HttpStatusCode.NotModified;
+                return;
+            }
+
+            HttpContext.Response.Headers[Constants.Headers.Etag] = CharExtensions.ToInvariantString(result.ResultEtag);
```

Comment:
I know that this is copied from other methods, but is this a proper format for Etag?
Check: https://github.com/ravendb/ravendb/blob/v4.0/src/Raven.Server/Web/System/StudioHandler.cs#L220